src/storage/RoutingResourceStore.ts

```diff
@@ -53,7 +53,7 @@ export class RoutingResourceStore implements ResourceStore {
     try {
       return await this.rule.handleSafe({ identifier });
     } catch (error: unknown) {
-      if (error instanceof UnsupportedHttpError) {
+      if (error instanceof BadRequestHttpError) {
         throw new NotFoundHttpError();
```

Comment:
Ooh what's this??

Reply:
This is an assumption that a router is going to throw a `BadRequestHttpError` if no matching store was found. And in that case we return a 404 if a GET was used instead.

Reply:
Sounds to me like we need a `NotImplementedError` in the router?

Reply:
Is it? (I truly have no idea :D ). An agent trying to access a path that the router doesn't link to can also be seen as the agent having made a bad request right?

Reply:
If it is btw, we should probably also throw the 501 if a user does a GET?

Reply:
Well… the router says "I don't support this"; from that, we could conclude a 404 because we don't find a router that supports it.

Reply:
Currently that would mean a 404 gets thrown for a GET and a 501 for a POST/PUT. Not sure what the best way to go is there then. Also 404?